src/map/skill.cpp

```diff
@@ -7711,6 +7710,12 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 			return 1;
 		}
 		status_damage(src, src, sstatus->max_hp,0,0,1, skill_id);
+		if(skill_id == HVAN_EXPLOSION && src->type == BL_HOM) {
+			TBL_HOM *hd = BL_CAST(BL_HOM,src);
+			hd->homunculus.intimacy = 100; // hom_intimacy_grade2intimacy(HOMGRADE_HATE_WITH_PASSION)
```

Comment:
```suggestion
			hd->homunculus.intimacy = hom_intimacy_grade2intimacy(HOMGRADE_HATE_WITH_PASSION);
```
Here it should be possible to use this since 100 and the value behind the function are the same.